Use Go embedding for YAML files instead of yamls2go

skitt opened the pull request:
See individual commits.

Fixes #1646.

skitt committed:
Switch to upstream-provided embedded MCS API CRDs

This reduces the risk of divergence between our local copy of the CRDs
and those provided by the SIG.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

tpantelis commented on deploy/mcsapi/crds/embed.go:
```diff
@@ -0,0 +1,29 @@
+
+	//go:embed multicluster.x_k8s.io_serviceimports.yaml
+	ServiceImportCRD []byte
+)
```
Perhaps we should wait for https://github.com/kubernetes-sigs/mcs-api/pull/115 and use the u/s CRDs...  Our copies match u/s.

skitt replied:
Yes, this is a draft to check the approach used in that PR.

skitt committed:
Ship all locally-maintained YAML files using go:embed

... instead of yamls2go. This avoids having to ensure the embedded
YAMLs are kept up-to-date.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt committed:
Import embedded YAMLs from Submariner

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt committed:
Embed operator CRDs

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt committed:
Remove embedded YAML framework

It is no longer used in the project.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt commented on Makefile:
```diff
@@ -148,17 +148,27 @@ CONTROLLER_DEEPCOPY := api/v1alpha1/zz_generated.deepcopy.go
+                   deploy/crds/submariner.io_brokers.yaml \
+				   deploy/crds/submariner.io_submariners.yaml \
+                   deploy/submariner/crds/submariner.io_clusterglobalegressips.yaml \
+				   deploy/submariner/crds/submariner.io_clusters.yaml \
```
These should be extracted from the Submariner constants if they’re still needed for the bundle.

skitt commented:
Remember this @Jaanki? :wink:

yboaron committed:
Merge branch 'devel' into embed-mcsapi-crds
